bin/agitator

```diff
@@ -30,6 +30,144 @@ Possible commands:
 EOF
 }
 
+function run_agitators() {
+  unset APP_HOSTS
+  export APP_HOSTS=$MANAGER_HOSTS
+  echo "Starting manager agitation as ${1} for ${APP_HOSTS}"
+  start_app_agitator "manager" "${AGTR_MASTER_KILL_SLEEP_TIME}" "${AGTR_MASTER_RESTART_SLEEP_TIME}" 1 1 > "${manager_log}" 2>&1 &
+  unset APP_HOSTS
+  export APP_HOSTS=$TSERVER_HOSTS
+  echo "Starting tserver agitation as ${1} for ${APP_HOSTS}"
+  start_app_agitator "tserver" "${AGTR_TSERVER_KILL_SLEEP_TIME}" "${AGTR_TSERVER_RESTART_SLEEP_TIME}" "${AGTR_TSERVER_MIN_KILL}" "${AGTR_TSERVER_MAX_KILL}" > "${tserver_log}" 2>&1 &
+}
+
+# Starts a app specific agitator
+# usage: start_app_agitator app_name kill_sleep_time restart_sleep_time min_kill max_kill
+# Requires that a list of hosts to be set in $APP_HOSTS
+function start_app_agitator() {
+  app_name=$1
+  kill_sleep_time=$2
+  restart_sleep_time=$3
+  min_kill=$4
+  max_kill=$5
+  ENV_VARS="ACCUMULO_HOME=${ACCUMULO_HOME} ZOOKEEPER_HOME=${ZOOKEEPER_HOME} HADOOP_HOME=${HADOOP_HOME} JAVA_HOME=${JAVA_HOME}"
+  if [[ -z $APP_HOSTS ]]; then
+    echo "ERROR: No hosts were found in env for ${app_name}"
+    exit 1
+  fi
+  T="$(date +'%Y%m%d %H:%M:%S')"
+  echo "${T} Running ${app_name} agitator ${kill_sleep_time} ${restart_sleep_time} ${min_kill} ${max_kill} for HOSTS=${APP_HOSTS}"
+  while true; do
+    echo "${T} Sleeping for ${kill_sleep_time} minutes"
+    sleep $((kill_sleep_time * 60))
+
+    T="$(date +'%Y%m%d %H:%M:%S')"
+    numToWack=$((min_kill + RANDOM % max_kill))
+    nodeToWack=$(echo "${APP_HOSTS}" | cut -d" " -f1)
+    hostsArray=("${APP_HOSTS}")
```

Comment:
don't forget to use local variables for anything you don't think should be set outside this function; if assigning a simple value to another local variable, you can do `local x=$y`. For arrays, you can do `local x=("${y[@]}")`. For assigning the result of an operation, you have to declare as local and assign it in two steps: `local x; x=$(echo y)`

My previous suggestion already set APP_HOSTS to an array from the function's input arguments, so that could simplify some of this array logic in here.